docs/tutorial.md

```diff
@@ -0,0 +1,159 @@
+#  kpack Tutorial
+
+This tutorial will walk through creating a kpack [image](image.md) resource to build a docker image from source and allow kpack to keep the image upto date.  
+
+1. Make sure you have kpack installed on your cluster and have access via kubectl
+
+    Note: Follow these docs to install kpack. 
+
+1. Create a secret with push credentials for the docker registry that you plan on publishing images to with kpack.  
+
+    ```yaml
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: registry-credentials
+      annotations:
+        build.pivotal.io/docker: <registry>
+    type: kubernetes.io/basic-auth
+    stringData:
+      username: <username>
+      password: <password>
+    ```
+   
+   Note: The secret must be annotated with the registry prefix for its corresponding registry. For [dockerhub](https://hub.docker.com/) this should be `index.docker.io`. 
+   For gcr this should be `gcr.io`.
+   
+   Your secret configuration should look something like this:
+   
+   ```yaml
```

Comment:
Not sure exactly why, but this renders with an extra space in front of each line, which fails to `kubectl apply` when copy/pasting. It will give a frustrating and obscure YAML error.